### PR TITLE
Add Nft Api Compute Rarity

### PR DESCRIPTION
**Pull request by xeno097**
### Changelog

- Adds the `computeRarity` method in the `nft` namespace.
- Adds the `NftAttributeRarity` type to represent the result of the `computeRarity` endpoint.
- Fixes a typo in the `OwnedBaseNftsResponse` type doc.
- Adds unit and integration tests for the `computeRarity` method.

### Extra
I was considering wrapping the `computeRarity` endpoint result in an object with the following structure `{attributes: NftAttributeRarity[] }` and exposing that in the public API but then decided to keep the same format as the endpoint response. Also, I'll gladly accept suggestions for the `NftAttributeRarity` type name if you have a better name for this use case.
The other endpoints listed in #140 will be added in another pr.

### CODE REVIEW

File: src/types/types.ts [hunk 1]
```diff
@@ -760,6 +760,22 @@ export interface GetFloorPriceResponse {
...
+ * Information about the rarity of an NFT's attribute in the specified collection.
+ *
+ * @public
+ */
+export interface NftAttributeRarity {
+  /** Name of the NFT's attribute. */
+  value: string;
+
+  /** The type of NFT attribute. */
+  trait_type: string;
```
Comment (thebrianchen) [inline src/types/types.ts:773]: Please rename to `traitType` in order to be consistent with JS style.

File: src/types/types.ts [hunk 2]
```diff
@@ -760,6 +760,22 @@ export interface GetFloorPriceResponse {
...
+
+  /** The NFT's attribute frequency in the current collection. */
```
Comment (thebrianchen) [inline src/types/types.ts:775]: Please update comment to:
`A number from 0 to 1 representing the prevalence of this value for this trait type in the current collection.`

File: test/integration/nft.test.ts [hunk 3]
```diff
@@ -240,6 +240,16 @@ describe('E2E integration tests', () => {
...
   });
 
+  it('computeRarity()', async () => {
+    const contractAddress = '0x0510745d2ca36729bed35c818527c4485912d99e';
+    const tokenId = '403';
+
+    const response = await alchemy.nft.computeRarity(contractAddress, tokenId);
+
+    expect(response).toBeDefined();
+    expect(response.length).toBeGreaterThan(0);
```
Comment (thebrianchen) [inline test/integration/nft.test.ts:250]: Please update the test to check that each field in the response type is present in the first element of the array.

File: test/unit/nft-api.test.ts [hunk 4]
```diff
@@ -1203,6 +1204,47 @@ describe('NFT module', () => {
...
+
+    it('calls with the correct parameters', async () => {
+      await alchemy.nft.computeRarity(contractAddress, tokenId);
+
+      expect(mock.history.get.length).toEqual(1);
+      expect(mock.history.get[0].params).toHaveProperty(
+        'contractAddress',
+        contractAddress
+      );
+      expect(mock.history.get[0].params).toHaveProperty('tokenId', tokenId);
```
Comment (thebrianchen) [inline test/unit/nft-api.test.ts:1235]: Once you update `trait_type` to `traitType`, can you also add a test to check that the method converts the field properly?

**Review by thebrianchen — CHANGES_REQUESTED** [review]
Thanks for taking this on -- we really appreciate it! Also thanks for catching some typos :)

Left a few nits and comments, but this is really good 🔥

File: src/types/types.ts (hunk 1 above)
Comment (xeno097) [reply src/types/types.ts:774]: On it 👌🏼

File: src/types/types.ts (hunk 2 above)
Comment (xeno097) [reply src/types/types.ts:776]: Okis, much better

File: test/unit/nft-api.test.ts (hunk 4 above)
Comment (xeno097) [reply test/unit/nft-api.test.ts:1236]: Sure, no problem.

File: test/integration/nft.test.ts (hunk 3 above)
Comment (xeno097) [reply test/integration/nft.test.ts:250]: Thank you for the suggestion!

File: test/unit/nft-api.test.ts [hunk 5]
```diff
@@ -1204,6 +1209,54 @@ describe('NFT module', () => {
     });
   });
 
+  describe('computeRarity()', () => {
+    const contractAddress = '0xbc4ca0eda7647a8ab7c2061c2e118a18a936f13d';
+    const tokenId = '7495';
```
Comment (thebrianchen) [inline test/unit/nft-api.test.ts:1214]: one more request: can you use a numerical value for `tokenId` here to check that the logic to convert `Bignumberish` input to a string is properly executed? Thanks!

**Review by thebrianchen — APPROVED** [review]
Thanks for the clean PR and fast turnaround times! Left one last request, but will merge this in after that 🚀

File: test/unit/nft-api.test.ts (hunk 5 above)
Comment (xeno097) [reply test/unit/nft-api.test.ts:1214]: Yup!

**Comment by xeno097** [comment]
> Thanks for the clean PR and fast turnaround times! Left one last request, but will merge this in after that rocket

🔥

File: test/unit/nft-api.test.ts (hunk 5 above)
Comment (xeno097) [reply test/unit/nft-api.test.ts]: Ready 👌🏼

**Pull request merged**